app/controllers/users_controller.rb

```diff
@@ -1,12 +1,20 @@
 class UsersController < ApplicationController
-  def new
-    @user = User.new
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
+
+  def index
+    @users = User.paginate page: params[:page]
   end
 
   def show
     @user = User.find_by id: params[:id]
```

Comment:
duplicate @user = User.find_by id: params[:id] in edit, update, destroy. You can define a method load_user and use it to 4 action like correct_user